app/models/katello/concerns/http_proxy_extensions.rb

```diff
@@ -16,6 +17,34 @@ def self.default_global_content_proxy
         end
       end
 
+      def repositories_with_proxy(proxy_id)
+        root_repos = RootRepository.with_selected_proxy(proxy_id)
+
+        if self == HttpProxy.default_global_content_proxy
+          root_repos += RootRepository.with_global_proxy
+        end
+
+        root_repos
+      end
+
+      def update_references_to_proxy
```

Comment:
small nitpick, but i think 'remove_references_to_proxy' would be more clear what its doing?

Reply:
I concur!